haystack/document_store/faiss.py

```diff
@@ -274,55 +299,63 @@ def train_index(
         :return: None
         """
 
+        index = index or self.index
         if embeddings and documents:
             raise ValueError("Either pass `documents` or `embeddings`. You passed both.")
         if documents:
             document_objects = [Document.from_dict(d) if isinstance(d, dict) else d for d in documents]
             doc_embeddings = [doc.embedding for doc in document_objects]
             embeddings_for_train = np.array(doc_embeddings, dtype="float32")
-            self.faiss_index.train(embeddings_for_train)
+            self.faiss_indexes[index].train(embeddings_for_train)
         if embeddings:
-            self.faiss_index.train(embeddings)
+            self.faiss_indexes[index].train(embeddings)
 
     def delete_all_documents(self, index: Optional[str] = None, filters: Optional[Dict[str, List[str]]] = None):
         """
         Delete all documents from the document store.
         """
+        if filters:
+            raise Exception("filters are supported for deleting documents in FAISSDocumentStore.")
         index = index or self.index
-        self.faiss_index.reset()
-        super().delete_all_documents(index=index, filters=filters)
-
-    def query_by_embedding(self,
-                           query_emb: np.ndarray,
-                           filters: Optional[dict] = None,
-                           top_k: int = 10,
-                           index: Optional[str] = None,
-                           return_embedding: Optional[bool] = None) -> List[Document]:
+        if index in self.faiss_indexes.keys():
+            self.faiss_indexes[index].reset()
+        super().delete_all_documents(index=index)
+
+    def query_by_embedding(
+        self,
+        query_emb: np.ndarray,
+        filters: Optional[Dict[str, List[str]]] = None,
+        top_k: int = 10,
+        index: Optional[str] = None,
+        return_embedding: Optional[bool] = None
+    ) -> List[Document]:
         """
         Find the document that is most similar to the provided `query_emb` by using a vector similarity metric.
 
         :param query_emb: Embedding of the query (e.g. gathered from DPR)
         :param filters: Optional filters to narrow down the search space.
                         Example: {"name": ["some", "more"], "category": ["only_one"]}
         :param top_k: How many documents to return
-        :param index: (SQL) index name for storing the docs and metadata
+        :param index: Index name to query the document from.
         :param return_embedding: To return document embedding
         :return:
         """
         if filters:
             raise Exception("Query filters are not implemented for the FAISSDocumentStore.")
-        if not self.faiss_index:
+
+        index = index or self.index
+        if not self.faiss_indexes.get(index):
             raise Exception("No index exists. Use 'update_embeddings()` to create an index.")
```

Comment:
Let's add the name of the index to the exception. It might be that only this index does not exist